addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/handler/FroniusSymoInverterHandler.java

```diff
@@ -74,7 +74,8 @@ public void initialize() {
     protected Object getValue(String channelId) {
         String[] fields = StringUtils.split(channelId, "#");
 
-        if (inverterRealtimeResponse == null) {
+        if (inverterRealtimeResponse == null || inverterRealtimeResponse.getBody() == null
+                || inverterRealtimeResponse.getBody().getData() == null) {
```

Comment:
This won't fix the problem. Data is not null in the json response but an empty object in case of an error returned. (and explains the recent reports this binding still crashes) So data will be an object, but all values in that object will not be set. Better solution would be to check the state value. The status message can then also be reported to the user.
To make the code more robust. All json representing objects can made to set and return an empty instance in case the value is null.

Reply:
Shouldn't there a Check for the status enough.
Somehting like:
```java
if(inverterRealtimeResponse.getHead().getStatus().getCode() == 0) {
 // everything is alright, go ahead
} 
else {
 // Something is wrong
}
```